scripts/main/photo.js

```diff
@@ -2,60 +2,59 @@
  * @description Takes care of every action a photo can handle and execute.
  */
 
-let photo = {
+const photo = {
+	/** @type {?Photo} */
 	json: null,
 	cache: null,
+	/** @type {?boolean} indicates whether the browser supports prefetching of images; `null` if support hasn't determined yet */
 	supportsPrefetch: null,
-	LivePhotosObject: null,
+	/** @type {?LivePhotosKit.Player} */
+	livePhotosObject: null,
 };
 
+/**
+ * @returns {?string} - the photo ID
+ */
 photo.getID = function () {
-	let id = null;
-
-	if (photo.json) id = photo.json.id;
-	else id = $(".photo:hover, .photo.active").attr("data-id");
+	let id = photo.json ? photo.json.id : $(".photo:hover, .photo.active").attr("data-id");
+	id = typeof id === "string" && /^[-_0-9a-zA-Z]{24}$/.test(id) ? id : null;
 
-	if ($.isNumeric(id) === true) return id;
-	else return false;
+	return id;
 };
 
+/**
+ *
+ * @param {string} photoID
+ * @param {string} albumID
+ * @param {boolean} autoplay - automatically start playback, if the photo is a video or live photo
+ *
+ * @returns {void}
+ */
 photo.load = function (photoID, albumID, autoplay) {
 	const checkContent = function () {
 		if (album.json != null && album.json.photos) photo.load(photoID, albumID, autoplay);
 		else setTimeout(checkContent, 100);
 	};
 
-	const checkPasswd = function () {
-		if (password.value !== "") photo.load(photoID, albumID, autoplay);
-		else setTimeout(checkPasswd, 200);
-	};
-
+	// TODO: The comment below sounds suspicious, what exactly is going on here?
 	// we need to check the album.json.photos because otherwise the script is too fast and this raise an error.
 	if (album.json == null || album.json.photos == null) {
 		checkContent();
-		return false;
+		return;
 	}
 
-	let params = {
-		photoID,
-		password: password.value,
-	};
-
-	api.post("Photo::get", params, function (data) {
-		if (data === "Warning: Photo private!") {
-			lychee.content.show();
-			lychee.goto();
-			return false;
-		}
-
-		if (data === "Warning: Wrong password!") {
-			checkPasswd();
-			return false;
-		}
-
+	/**
+	 * @param {Photo} data
+	 * @returns {void}
+	 */
+	const successHandler = function (data) {
 		photo.json = data;
-		photo.json.original_album = photo.json.album;
-		photo.json.album = albumID;
+		// TODO: `photo.json.original_album_id` is set only, but never read; do we need it?
```

Comment:
I've wondered the same thing... In principle it feels like we _should_ be using it (e.g., to determine the permissions of photos from Recent) but we don't...